### ISSUE TITLE
stuck during training

### ISSUE THREAD

**Issue by world4jason** [HRNet/HRNet-Semantic-Segmentation#26]
I download the pretrained_models and modified GPU setting from (0,1,2,3) to (0,)
but the training process stuck at here

> Total Parameters: 65,773,843
>
> ----------------------------------------------------------------------------------------------------------------------------------
>Total Multiply Adds (For Convolution and Linear Layers only): 174.0439453125 GFLOPs
>
>----------------------------------------------------------------------------------------------------------------------------------
>Number of Layers
>Conv2d : 307 layers   InPlaceABNSync : 306 layers   ReLU : 269 layers   Bottleneck : 4 layers   BasicBlock : 104 layers   HighResolutionModule : 8 layers`

any idea about how this happened?

**Comment by world4jason** [comment]
sorry for the foggy question above.
I mean it seems not thing shows in tensorboard nor any kind of progress bar indicate that i'm in the training proccess

**Comment by sunke123** [comment]
You can change the PRINT_FREQ to 1 in the yaml file.
I wonder if one iteration cost too much time.

**Comment by zhixuanli** [comment]
I have set PRINT_FREQ to 1, but still get stucked at here:
```
Total Parameters: 65,773,843
----------------------------------------------------------------------------------------------------------------------------------
Total Multiply Adds (For Convolution and Linear Layers only): 174.0439453125 GFLOPs
----------------------------------------------------------------------------------------------------------------------------------
Number of Layers
Conv2d : 307 layers   InPlaceABNSync : 306 layers   ReLU : 269 layers   Bottleneck : 4 layers   BasicBlock : 104 layers   HighResolutionModule : 8 layers   

```
@sunke123 Can you give me some advice? Thank you!

**Comment by Hussainflr** [comment]
I'm also experiencing the same problem, has anybody got a solution? why it gets stuck. I have tried on both datasets LIP and Cityscapes but got the same problem.

**Comment by chipkajb** [comment]
I am experiencing the same problem also. It looks like it happens on the `nn.parallel.DistributedDataParallel` line in train.py. Does anyone know how to fix this?

**Comment by chipkajb** [comment]
It appears to be working for me now. I tried changing many different things, but I think the root of the problem might have been insufficient shared memory. I am running inside a docker container, so I just really cranked up the accessible shared memory by adding `--shm-size=128g` to my docker run command and that seemed to do the trick.

**Comment by sharatg9** [comment]
> I have set PRINT_FREQ to 1, but still get stucked at here:
> 
> ```
> Total Parameters: 65,773,843
> ----------------------------------------------------------------------------------------------------------------------------------
> Total Multiply Adds (For Convolution and Linear Layers only): 174.0439453125 GFLOPs
> ----------------------------------------------------------------------------------------------------------------------------------
> Number of Layers
> Conv2d : 307 layers   InPlaceABNSync : 306 layers   ReLU : 269 layers   Bottleneck : 4 layers   BasicBlock : 104 layers   HighResolutionModule : 8 layers   
> ```
> 
> @sunke123 Can you give me some advice? Thank you!

any fix for this ??

**Comment by tjzjp** [comment]
> I download the pretrained_models and modified GPU setting from (0,1,2,3) to (0,)
> but the training process stuck at here
> 
> > Total Parameters: 65,773,843
> > 
> > Total Multiply Adds (For Convolution and Linear Layers only): 174.0439453125 GFLOPs
> > 
> > Number of Layers
> > Conv2d : 307 layers   InPlaceABNSync : 306 layers   ReLU : 269 layers   Bottleneck : 4 layers   BasicBlock : 104 layers   HighResolutionModule : 8 layers`
> 
> any idea about how this happened?

I met the same problem, I think probabably it stuck in BN layer.
I am still struggle in this problem, anyone has a solution?

**Comment by sharatg9** [comment]
> > I download the pretrained_models and modified GPU setting from (0,1,2,3) to (0,)
> > but the training process stuck at here
> > > Total Parameters: 65,773,843
> > > Total Multiply Adds (For Convolution and Linear Layers only): 174.0439453125 GFLOPs
> > > Number of Layers
> > > Conv2d : 307 layers   InPlaceABNSync : 306 layers   ReLU : 269 layers   Bottleneck : 4 layers   BasicBlock : 104 layers   HighResolutionModule : 8 layers`
> > 
> > 
> > any idea about how this happened?
> 
> I met the same problem, I think probabably it stuck in BN layer.
> I am still struggle in this problem, anyone has a solution?

hey, that's correct it stuck in bn layer. 
just change the bn layer to ABN. it will work for single gpu as well. 

@seg_henrt.py
BatchNorm2d = functools.partial(**ABN**, activation='none')

but i m not sure it gives you the same performance in terms of accuracy

**Comment by tjzjp** [comment]
It really can work！
Thank you very much！




    


On 03/20/2020 18:40, sharatg9 wrote: 


I download the pretrained_models and modified GPU setting from (0,1,2,3) to (0,)
but the training process stuck at here

Total Parameters: 65,773,843
Total Multiply Adds (For Convolution and Linear Layers only): 174.0439453125 GFLOPs
Number of Layers
Conv2d : 307 layers   InPlaceABNSync : 306 layers   ReLU : 269 layers   Bottleneck : 4 layers   BasicBlock : 104 layers   HighResolutionModule : 8 layers`

any idea about how this happened?

I met the same problem, I think probabably it stuck in BN layer.
I am still struggle in this problem, anyone has a solution?

hey, that's correct it stuck in bn layer.
just change the bn layer to ABN. it will work for single gpu as well.
@seg_henrt.py
BatchNorm2d = functools.partial(ABN, activation='none')
but i m not sure it gives you the same performance in terms of accuracy

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/HRNet/HRNet-Semantic-Segmentation/issues/26#issuecomment-601633829",
"url": "https://github.com/HRNet/HRNet-Semantic-Segmentation/issues/26#issuecomment-601633829",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
